scripts/tableland/tasks.ts

```diff
@@ -69,5 +69,108 @@ task('create-dd-table', 'Creates a Device Definition table related to a Manufact
+
+        console.log('Get device definitions...');
+        const devices = (await getDeviceDefinitions()).data.device_definitions;
+        console.log(`Total device definitions ${devices.length}...`);
+
+        // manufacturers = manufacturers.slice(0, 5);
+        for await (const element of manufacturers) {
+            const name = `${element.name_slug}`;
+
+            console.log(`Creating Manufacturer ${name} ...`);
+
+            const tx = await manufacturerInstance
+                .connect(signer)
+                .mintManufacturerBatch(signer.address, [name]);
+
+            const receipt = await tx.wait();
+            const manufacturerId = (receipt?.logs[1] as EventLog).args[1].toString();
+
+            console.log(`Manufacturer ${name} minted with ID: ${manufacturerId}`);
+
+            const ddTableInstance: DeviceDefinitionTable = await hre.ethers.getContractAt(
+                'DeviceDefinitionTable',
+                instances.localhost.modules.DIMORegistry.address,
+            );
+
+            console.log(`Creating Device Definition table for manufacturer ${name} with ID  ${manufacturerId}...`);
+
+            const dTx = await ddTableInstance
+                .connect(signer)
+                .createDeviceDefinitionTable(tableOwner, manufacturerId, name);
+
+            await tablelandValidator.pollForReceiptByTransactionHash({
+                chainId: 31337,
+                transactionHash: (await dTx.wait())?.hash as string,
+            });
+
+            const ddTableId = await ddTableInstance.getDeviceDefinitionTableId(manufacturerId);
+            const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId, name);
+
+            console.log(`Device Definition table created\nTable ID: ${ddTableId}\nTable Name: ${ddTableName}`);
+
+            const deviceDefinitionByManufacturers = devices.filter((c)=> c.make.name_slug === element.name_slug && c.type.year > 2006);
+            console.log(`Get Device Definition By Manufacturer [${element.name}] total => ${deviceDefinitionByManufacturers.length}`);
+
+            // deviceDefinitionByManufacturers = deviceDefinitionByManufacturers.slice(0, 5);
+            for await (const dd of deviceDefinitionByManufacturers) {
+                
+                const deviceDefinitionInput : DeviceDefinitionInput = {
+                    id: `${dd.make.name_slug}-${dd.type.model_slug}-${dd.type.year}`,
+                    deviceDefinitionId: dd.device_definition_id,
+                    model: dd.type.model,
+                    year: dd.type.year,
+                    metadata: JSON.stringify({
+                        vehicle_data: dd.vehicle_data,
+                        device_attributes: dd.device_attributes,
+                        device_integrations: dd.device_integrations
+                    })
+                };
+
+                await ddTableInstance.insertDeviceDefinition(manufacturerId, name, deviceDefinitionInput);
```

Comment:
We have a `insertDeviceDefinitionBatch` function. Might be more efficient to used. We can start trying batches of 100 or 50. I still have to calculate how it would cost.

Reply:
Fixed!!